StaySphere.Api/StaySphere.Infrastructure/Persistence/Repositories/CommonRepository.cs

```
@@ -0,0 +1,104 @@
+                return _booking;
+            }
+        }
+
+        public ICategoryRepository _category;
+        public ICategoryRepository Category
+        {
+            get
+            {
+                _category ??= new CategoryRepository(_context);
+                return _category;
+            }
+        }
+
+        public IDocumentRepository _document;
+        public IDocumentRepository Document
+        {
+            get
+            {
+                _document ??= new DocumentRepository(_context);
+                return _document;
+            }
+        }
+
+        public IEmployeeRepository _employee;
+        public IEmployeeRepository Employee
+        {
+            get
+            {
+                _employee ??= new EmployeeRepository(_context);
+                return _employee;
+            }
+        }
+
+        public IGuestRepository _guest;
+        public IGuestRepository Guest
+        {
+            get
+            {
+                _guest ??= new GuestRepository(_context);
+                return _guest;
+            }
+        }
+
+        public IPositionRepository _position;
+        public IPositionRepository Position
+        {
+            get
+            {
+                _position ??= new PositionRepository(_context);
+                return _position;
+            }
+        }
+
+        public IReviewRepository _review;
+        public IReviewRepository Review
+        {
+            get
+            {
+                _review ??= new ReviewRepository(_context);
+                return _review;
+            }
+        }
+
+        public IRoomRepository _room;
+        public IRoomRepository Room
+        {
+            get
+            {
+                _room ??= new RoomRepository(_context);
+                return _room;
+            }
+        }
+
+        public CommonRepository(StaySphereDbContext context)
+        {
+            _context = context;
```

Should we throw an exception of context is null?